OSDOCS-2602: Improve CatalogSource concept docs

adellape opened the pull request:
https://issues.redhat.com/browse/OSDOCS-2602

* Updated OLM's global namespace to be a replaceable attribute based on upstream (the `olm` namespace for the `openshift-origin` distro) vs downstream (the `openshift-marketplace` namespace for any other distro).
* Re-wrote/re-org'd content from intro para
* Added `[TIP]` admonition
* Updated the example YAML and added more callout descriptions along the way
* Added `status` section in YAML example
* Added "Additional resources"
* Added "Viewing Operator catalog source status by using the CLI"

Preview: 

* [Catalog source](https://deploy-preview-35853--osdocs.netlify.app/openshift-enterprise/latest/operators/understanding/olm/olm-understanding-olm.html#olm-catalogsource_olm-understanding-olm)
* [Viewing Operator catalog source status using the CLI](https://deploy-preview-35853--osdocs.netlify.app/openshift-enterprise/latest/operators/admin/olm-status.html#olm-cs-status-cli_olm-status) (also appears in the [Troubleshooting Operator issues](https://deploy-preview-35853--osdocs.netlify.app/openshift-enterprise/latest/support/troubleshooting/troubleshooting-operator-issues.html) doc in the _Support_ guide)

Xia-Zhao-rh commented:
/lgtm

dmesser commented:
This looks good. I think some additional hints around the pod backing the catalog and troubleshooting that in case the catalog status fails to transition to `READY` would be great.

adellape commented:
@dmesser @Xia-Zhao-rh

> I think some additional hints around the pod backing the catalog and troubleshooting that in case the catalog status fails to transition to `READY` would be great.

Added a [Viewing Operator catalog source status using the CLI](https://deploy-preview-35853--osdocs.netlify.app/openshift-enterprise/latest/operators/admin/olm-status.html#olm-cs-status-cli_olm-status) section to match the existing [subscription status-related procedure](https://deploy-preview-35853--osdocs.netlify.app/openshift-enterprise/latest/operators/admin/olm-status.html#olm-status-viewing-cli_olm-status), which is now linked to from the "Catalog source" concept page.

Overall diff of changes since last review:

https://github.com/openshift/openshift-docs/compare/6c7cff5d511d42fd65f7fa434f7b7109782046a9..52b870783faee107232f4267cbfcd6ab3d236c97

mikemckiernan commented on modules/olm-catalogsource.adoc:
```diff
@@ -2,52 +2,100 @@
 ----
-<1> Specify catalog image.
-<2> Automatically check for new versions at a given interval to keep up to date.
+<.> Name for the `CatalogSource` object. This name is also used to create the name for the related pod that is created in the requested namespace.
```
nit: Can you avoid "name is also used to create the name"?

This value is also used to name the related pod...

Or, because I don't know what I'm typing about:
This value is used as part of the name for the related pod...

mikemckiernan commented on modules/olm-catalogsource.adoc:
```diff
@@ -2,52 +2,100 @@
-<1> Specify catalog image.
-<2> Automatically check for new versions at a given interval to keep up to date.
+<.> Name for the `CatalogSource` object. This name is also used to create the name for the related pod that is created in the requested namespace.
+<.> Namespace to make the catalog available in. To make the catalog available cluster-wide in all namespaces, set this value to `{global_ns}`. The default Red Hat-provided catalog sources also use the `{global_ns}` namespace.
```
maybe s/make/create/ the first instance.

because that's the verb used with Kubernetes and for no other reason whatsoever.

(I thought the ISG had guidance to avoid hyphens with company names, but I must have imagined it.)

mikemckiernan commented on modules/olm-catalogsource.adoc:
```diff
@@ -2,52 +2,100 @@
+<.> Namespace to make the catalog available in. To make the catalog available cluster-wide in all namespaces, set this value to `{global_ns}`. The default Red Hat-provided catalog sources also use the `{global_ns}` namespace.
+<.> Display name for the catalog in the web console and CLI.
+<.> Index image for the catalog.
+<.> Weight for the catalog source. OLM uses the weight for prioritization during dependency resolution. Higher weight indicates the catalog is preferred over lower weighted catalogs.
```
maybe "A higher weight indicates..." and maybe "over lower-weighted catalogs."

mikemckiernan commented on modules/olm-cs-status-cli.adoc:
```diff
@@ -0,0 +1,119 @@
+endif::[]
+
+[id="olm-cs-status-cli_{context}"]
+= Viewing Operator catalog source status using the CLI
```
s/using/by using/ kind of everywhere

mikemckiernan commented on modules/olm-cs-status-cli.adoc:
```diff
@@ -0,0 +1,119 @@
+
+.Procedure
+
+. List the catalog sources for the namespace where your catalog source was created. For example, you can check the `{global_ns}` namespace, which is used for cluster-wide catalog sources:
```
The repetition of "catalog source" in the first sentence seems a little awk.  Obligatory sugg:

. List the catalog sources in a namespace. For example,...

If you prefer the orig, maybe "...where you created the catalog source."

mikemckiernan commented on modules/olm-cs-status-cli.adoc:
```diff
@@ -0,0 +1,119 @@
+    Service Namespace:  {global_ns}
+----
++
+In this example output, the last observed state is `TRANSIENT_FAILURE`, which indicates that there is a problem establishing a connection for the catalog source.
```
up to you on a split: "...`TRANSIENT_FAILURE`. This state indicates a problem..."

mikemckiernan commented on modules/olm-cs-status-cli.adoc:
```diff
@@ -0,0 +1,119 @@
++
+In this example output, the last observed state is `TRANSIENT_FAILURE`, which indicates that there is a problem establishing a connection for the catalog source.
+
+. When a catalog source is created in a namespace, a pod for the catalog source is created in that namespace. To get more details about catalog source pods, list the pods in the namespace where your catalog source was created:
```
I feel compelled to suggest starting the step with a verb.  Can you "List the pods in the namespace:" and then put the information about the purpose of the pod after the output?  Just an idea.

mikemckiernan commented on modules/olm-cs-status-cli.adoc:
```diff
@@ -0,0 +1,119 @@
+redhat-operators-smxx8                  1/1     Running            0          36m
+----
++
+In this example output, the status for the `example-catalog-bwt8z` pod is `ImagePullBackOff`, which indicates that there is an issue pulling the catalog source's index image.
```
maybe "In the preceding example output..."

Up to you on a sentence split: "...`ImagePullBackOff`. This status indicates an issue with.."

mikemckiernan commented on modules/olm-cs-status-cli.adoc:
```diff
@@ -0,0 +1,119 @@
+  Warning  Failed          8s (x3 over 47s)   kubelet            Error: ErrImagePull
+----
++
+In this example output, the error messages indicate that the catalog source's index image is failing to pull successfully because of an authorization issue. For example, the index image might be stored in a registry that requires login credentials.
```
Friendly reminder to repeat "In the preceding example output" or ignore the sugg.

mikemckiernan commented on modules/olm-subscription.adoc:
```diff
@@ -2,26 +2,33 @@
 = Subscription
 
-A subscription, defined by a `Subscription` object, represents an intention to install an Operator. It is the custom resource that relates an Operator to a catalog source.
+A _subscription_, defined by a `Subscription` object, represents an intention to install an Operator. It is the custom resource that relates an Operator to a catalog source.
```
Love these crisp sentences that folks can build the conceptual model on.

mikemckiernan reviewed (commented):
Hopefully the review helps.  Ignore anything you disagree with. If I goofed something, please let me know.

adellape commented:
Merging this so I can build off of it for the `catalogImageTemplate` docs in https://github.com/openshift/openshift-docs/pull/36199, but if there are any further suggestions/edits to come from this PR, I can follow-up there.

adellape commented:
/cherrypick enterprise-4.9
